### ISSUE TITLE
0.6.2 producing different computations vs 0.6.1 for mfccs, delta, zcr, rmse, spectral bandwidth, spectral centroids

### ISSUE THREAD

**Issue by brityboy** [librosa/librosa#792]
#### Description
I was running a project using the following features
 - mfccs
 - delta
 - zcr
 - rmse
 - spectral bandwidth
 - spectral centroid

Using the following below successfully
 - Darwin-16.7.0-x86_64-i386-64bit
 - Python 2.7.?? (It was the latest 64bit macOS Anaconda distribution as of April 23rd)
 - numpy 1.14.5
 - Scipy 1.1.0
 - librosa 0.6.1

Basically, bad things happened to my Anaconda distribution and I had to reinstall things on Oct 23rd

With the new environment of:
 - Darwin-16.7.0-x86_64-i386-64bit
 - Python 2.7.?? (It was the latest 64bit macOS Anaconda distribution as of April 23rd)
 - numpy 1.14.5
 - Scipy 1.1.0

I found that I had two problems
a. When installing librosa 0.6.1, I could no longer import it - I would always only get the error below
b. When computing features these features on librosa 0.6.2, I noticed that they were different mostly after 11 to 12 decimal places for ~15% of my dataset

Which is why I really wanted to just use librosa 0.6.1 again to see what's going on there - but I've been unable to import it successfully after.

This is true on two of my laptops, a macOS one and an Ubuntu 18, info for both environments below

#### Steps/Code to Reproduce
1. create a virtual environment venv
2. activate it and install librosa 0.6.1 via `pip install librosa==0.6.1`
3. open ipython and import librosa

#### Expected Results
librosa 0.6.1 should basically import successfully

#### Actual Results
```
/home/brian/Documents/projects/test.py in <module>()
----> 1 import librosa

/home/brian/Documents/projects/venv/lib/python2.7/site-packages/librosa/__init__.py in <module>()
     10 # And all the librosa sub-modules
     11 from . import cache
---> 12 from . import core
     13 from . import beat
     14 from . import decompose

/home/brian/Documents/projects/venv/lib/python2.7/site-packages/librosa/core/__init__.py in <module>()
    102 """
    103 
--> 104 from .time_frequency import *  # pylint: disable=wildcard-import
    105 from .audio import *  # pylint: disable=wildcard-import
    106 from .spectrum import *  # pylint: disable=wildcard-import

/home/brian/Documents/projects/venv/lib/python2.7/site-packages/librosa/core/time_frequency.py in <module>()
      8 import six
      9 
---> 10 from ..util.exceptions import ParameterError
     11 
     12 __all__ = ['frames_to_samples', 'frames_to_time',

/home/brian/Documents/projects/venv/lib/python2.7/site-packages/librosa/util/__init__.py in <module>()
     65 """
     66 
---> 67 from .utils import *  # pylint: disable=wildcard-import
     68 from .files import *  # pylint: disable=wildcard-import
     69 from .matching import *  # pylint: disable=wildcard-import

/home/brian/Documents/projects/venv/lib/python2.7/site-packages/librosa/util/utils.py in <module>()
    109 
    110 @cache(level=20)
--> 111 def valid_audio(y, mono=True):
    112     '''Validate whether a variable contains valid, mono audio data.
    113 

/home/brian/Documents/projects/venv/lib/python2.7/site-packages/librosa/cache.py in wrapper(function)
     47                     dict(decorated=dec(func)), __wrapped__=func)
     48 
---> 49             if self.cachedir is not None and self.level >= level:
     50                 return decorator_apply(self.cache, function)
     51 

/home/brian/Documents/projects/venv/lib/python2.7/site-packages/joblib/memory.pyc in cachedir(self)
    903             "and will be removed in version 0.14.\n"
    904             "Use os.path.join(memory.location, 'joblib') attribute instead.",
--> 905             DeprecationWarning, stacklevel=2)
    906         if self.location is None:
    907             return None

TypeError: expected string or buffer
```

#### Versions

macOS environment
Darwin-16.7.0-x86_64-i386-64bit
Python 2.7.15 Anaconda default as of Oct 23, 2018
audioread==2.1.6
decorator==4.3.0
enum34==1.1.6
funcsigs==1.0.2
joblib==0.12.0
librosa==0.6.1 (which can't be imported)
llvmlite==0.23.2
numba==0.38.1
numpy==1.14.5
resampy==0.2.1
scikit-learn==0.19.1
scipy==1.1.0
singledispatch==3.4.0.3
six==1.11.0

Ubuntu environment
Linux-4.15.0-36-generic-x86_64-with-debian-buster-sid
('Python', '2.7.14 |Anaconda, Inc.| (default, Dec  7 2017, 17:05:42) \n[GCC 7.2.0]')
('NumPy', '1.15.4')
('SciPy', '1.1.0')
audioread==2.1.6
decorator==4.3.0
enum34==1.1.6
funcsigs==1.0.2
joblib==0.13.0
librosa==0.6.1 (also can't be imported)
llvmlite==0.25.0
numba==0.40.1
numpy==1.15.4
resampy==0.2.1
scikit-learn==0.20.0
scipy==1.1.0
singledispatch==3.4.0.3
six==1.11.0

**Comment by bmcfee** [comment]
This is a known compatibility issue #729 that has already been fixed in the 0.6.2 release.

You can either upgrade librosa to 0.6.2, or pin joblib to 0.11.

> b. When computing features these features on librosa 0.6.2, I noticed that they were different mostly after 11 to 12 decimal places for ~15% of my dataset

What precision of floats were you using?  Differences in the 11th decimal place are pretty miniscule, and might be down to machine precision issues if you're in float32.  (See `np.finfo(np.float32)`: resolution is 1e-6 there, so any differences past that are going to be pretty unreliable.)

Otherwise, it may be due to differences in the BLAS implementation, llvmlite/numba, etc.  0.6.1-0.6.2 was a pretty minor revision, and we didn't change anything algorithmic, so I doubt the differences are due to librosa itself.  You can see [the changelog](https://librosa.github.io/librosa/changelog.html#v0-6-2) for details.

**Comment by brityboy** [comment]
Firstly, thanks for responding so quick even during holidays!

Moving to joblib 0.11 worked like a charm

As to precision, I had actually just let things run their course in the past - only discovering things now  that I'm trying to figure things out (meaning why they're different)

Meanwhile, I don't think I can get the original values so I'm closing this issue. Thanks!
